### PR TITLE
Update dependencies to Kubernetes v1.21.1

### PR DESCRIPTION
**Pull request by ash2k**
Mostly to drop old Kustomize and other old dependency versions so that this library plays nicer with downstream consumers that want to use more modern versions.

Edit: Switched all `go get` to `go install` because `go get` modifies `go.mod`/`go.sum` files while installing a package and `go install` does not. We want the later beacuse we're not adding the dependency, just installing something to use for code generation/etc. See https://golang.org/doc/go1.16#go-command

### CODE REVIEW

**Comment by ash2k** [comment]
Build is failing because Go 1.13 is used. I've opened https://github.com/kubernetes/test-infra/pull/22350 to update it.

File: Makefile
Comment (ash2k) [inline Makefile:38]: Pinned dependency versions

File: examples/alphaTestExamples/inventoryNamespace.md
Comment (ash2k) [inline examples/alphaTestExamples/inventoryNamespace.md:102]: There is a [new `ConfigMap` `kube-root-ca.crt`](https://v1-20.docs.kubernetes.io/docs/setup/release/notes/#introducing-rootcaconfigmap) now, so made this more robust.
```
NAME               DATA   AGE
cm-a               0      2s
cm-b               0      2s
cm-c               0      2s
kube-root-ca.crt   1      7s
```

Comment (ash2k) [reply examples/alphaTestExamples/inventoryNamespace.md:102]: Previously `kind` was creating Kubernetes v1.19.1 and now it creates v1.21.1.

File: go.mod
Comment (ash2k) [inline go.mod:21]: Otherwise it's resolved to an older version, which is incompatible. This version contains the `gnostic` bump from https://github.com/kubernetes/kube-openapi/pull/220. Kubernetes `master` uses gnostic `0.5.1` too.

File: test/e2e/serverside_apply_test.go
Comment (ash2k) [inline test/e2e/serverside_apply_test.go:54]: 1.19.1 didn't support this, 1.21.1 does.

**Comment by ash2k** [comment]
@mortent this is finally ready for review. Cheers.

**Comment by tamalsaha** [comment]
Should klog be also updated to klog/v2?

**Comment by mortent** [comment]
/lgtm
I'll let @seans3 take a look and do the final approval.

File: Makefile
Comment (seans3) [reply Makefile]: Technically running a version 1.21 `kubectl` against a 1.19 API Server is outside the support window. `kubectl` is guaranteed  to work with +/- 1 version, so a 1.21 `kubectl` (we wrap `kubectl apply`) is supported for 1.20 - 1.22.

I do not think we need to worry about this, but let's be on the look-out.

**Comment by seans3** [comment]
> Should klog be also updated to klog/v2?

You are right. `klog` is outdated and we should update it to the more recent `klog/v2`. I do not think it is currently working. Let's clean that up in another PR.

File: test/e2e/serverside_apply_test.go
Comment (seans3) [inline test/e2e/serverside_apply_test.go:68]: Adding a comment that `LastAppliedConfigAnnotation` is only set for client-side apply, and we've server-side applied here would make this easier for others to follow.

File: test/e2e/serverside_apply_test.go
Comment (seans3) [inline test/e2e/serverside_apply_test.go:50]: We didn't add sufficient comments initially, but we might want to comment somewhere about the difference of client-side (sets `LastAppliedConfigAnnotation`) vs. server-side apply (sets field managers) to help explain the tests.

**Review by seans3 — CHANGES_REQUESTED** [review]
Thanks for all the work. One small nitpick. I'll merge this quickly once we've added the comments. Let's be on the lookout for version skew issues.

/lgtm

**Comment by ash2k** [comment]
@seans3 Thanks for the review. Looks like the bot is buggy and merged it anyway. I'll open a followup with extra comments.

**Pull request merged**
